src/main/java/org/folio/circulation/domain/FeeAmount.java

```diff
@@ -0,0 +1,53 @@
+package org.folio.circulation.domain;
+
+import static org.folio.circulation.support.JsonPropertyFetcher.getBigDecimalProperty;
+
+import java.math.BigDecimal;
+
+import io.vertx.core.json.JsonObject;
+
+public final class FeeAmount {
+  private static final FeeAmount ZERO = new FeeAmount(BigDecimal.ZERO);
+
+  private final BigDecimal amount;
+
+  public FeeAmount(BigDecimal amount) {
+    this.amount = amount != null ? amount : BigDecimal.ZERO;
+  }
+
+  public FeeAmount(double amount) {
+    this(BigDecimal.valueOf(amount));
+  }
+
+  public final FeeAmount subtract(FeeAmount toSubtract) {
+    return new FeeAmount(amount.subtract(toSubtract.amount));
+  }
+
+  public final FeeAmount add(FeeAmount toAdd) {
+    return new FeeAmount(amount.add(toAdd.amount));
+  }
+
+  public final double toDouble() {
+    return scaledAmount().doubleValue();
+  }
+
+  public boolean hasAmount() {
+    return amount.compareTo(BigDecimal.ZERO) > 0;
+  }
+
+  private BigDecimal scaledAmount() {
+    return amount.setScale(2, BigDecimal.ROUND_HALF_UP);
+  }
```

Comment:
Yes, to prevent the fee amount like 10.123 etc.

Also, fee/fine module uses double for amount manipulations. Since double is not so accurate with fractional parts, it is possible that fee amount may have some "noise", e.g. 0.00000123 - it is not equal to zero, however it is a zero in terms of fee.

Reply:
Has this rounding decision been agreed with Holly?

Reply:
Good point, will ask her.

Reply:
Here is the question https://issues.folio.org/browse/CIRC-717?focusedCommentId=77199&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-77199

Reply:
Answer from Holly:

> Bohdan Suprun: _1. Half up means values >= 0.005 will be rounded to 0.01 and values < 0.005 will be rounded to 0.00 _makes the most sense to me. Thanks for checking...

Resolving as no changes needed.